flight/upload_mission.py

```diff
@@ -6,19 +6,25 @@
 import logging
 from mavsdk import System
 from mavsdk.mission import MissionItem, MissionPlan
+from flight import intake_gps
 
-from intake_gps import Waypoint, extract_gps
 
-
-async def upload_mission(path: str = "flight/data/target_data.json") -> None:
+async def upload_mission(competition: bool) -> None:
     """
     Connects to the drone and uploads the mission plan for landing the drone.
 
     Parameters
     ----------
-    path : str
-        File path to the target data JSON file.
+    competition : bool
+        Decides if competition waypoints are used in the mission or not.
     """
+
+    waypoint_path: str
+    if competition:
+        waypoint_path = "flight/data/target_data.json"
+    else:
+        waypoint_path = "flight/data/golf_target.json"
+
```

Comment:
Same thing in this file about `waypoint_path`. It's not really a path if it's just one point. Again, if I'm mistaken, just ignore this comment.